docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/schema/SchemasForEndpoints.scala

```diff
@@ -1,19 +1,24 @@
 package sttp.tapir.docs.apispec.schema
 
 import sttp.tapir.SchemaType.SObjectInfo
-import sttp.tapir.apispec.{Schema => ASchema, _}
 import sttp.tapir._
+import sttp.tapir.apispec.{Schema => ASchema, _}
 import sttp.tapir.internal.IterableToListMap
 
 import scala.collection.immutable.ListMap
 
-object SchemasForEndpoints {
-  def apply(es: Iterable[Endpoint[_, _, _, _]], schemaName: SObjectInfo => String): (ListMap[ObjectKey, ReferenceOr[ASchema]], Schemas) = {
-    val sObjects = ObjectSchema.unique(es.flatMap(e => forInput(e.input) ++ forOutput(e.errorOutput) ++ forOutput(e.output)))
+class SchemasForEndpoints(
+    es: Iterable[Endpoint[_, _, _, _]],
+    schemaName: SObjectInfo => String,
+    referenceEnums: SObjectInfo => Boolean = _ => false
```

Comment:
I'd omit defaults here as well. They are useful for user-facing APIs, but otherwise it's ease to miss a parameter that needs to be passed